src/PyEventLoop.cc

```diff
@@ -172,19 +196,27 @@ PyEventLoop PyEventLoop::getRunningLoop() {
 }
 
 void PyEventLoop::AsyncHandle::cancel() {
-  PyObject *scheduled = PyObject_GetAttrString(_handle, "_scheduled"); // this attribute only exists on asyncio.TimerHandle returned by loop.call_later
-                                                                       // NULL if no such attribute (on a strict asyncio.Handle returned by loop.call_soon)
-  bool finishedOrCanceled = scheduled && scheduled == Py_False; // the job function has already been executed or canceled
-  if (!finishedOrCanceled) {
+  if (!_finishedOrCancelled()) {
     removeRef(); // automatically unref at finish
   }
-  Py_XDECREF(scheduled);
 
   // https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.Handle.cancel
   PyObject *ret = PyObject_CallMethod(_handle, "cancel", NULL); // returns None
   Py_XDECREF(ret);
 }
 
+bool PyEventLoop::AsyncHandle::cancelled() {
+  // https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.Handle.cancelled
+  PyObject *ret = PyObject_CallMethod(_handle, "cancelled", NULL); // returns Python bool
+  return ret == Py_True;
```

Comment:
```suggestion
  bool retVal = ret == Py_True;
  Py_XDECREF(ret);
  return retVal;
```

Reply:
This is good to have, but not necessary, since `Py_True`/`Py_False` are [singletons](https://github.com/python/cpython/blob/3.12/Include/boolobject.h#L20-L22)

Reply:
Let's then put it in for the good-to-have part :-)

Reply:
https://github.com/Distributive-Network/PythonMonkey/pull/311/commits/aeec9a977c8820c9beb242706dfc71a97fefd3a5